fix: trigger warning toast when localstorage is full

mcendon opened the pull request:
Closes #318

* Added the warning toast with "Failed to save to localStorage" and error message when localStorage quota is full
* Playwright test added

mcendon committed:
fix: trigger warning toast when the localStorage is full (#318)

mcendon committed:
test: warning toast should be shown when localStorage is full #318

fmaclen reviewed (changes requested):
Thanks for submitting the PR, left you some minor notes.

mcendon committed:
test,fix: improve localStorage is full test and enhance toast user-friendly message (#318)

fmaclen reviewed (changes requested):
I think we are more or less there. Found a styling bug with the toast, it's rendered in a semi-transparent blue color:

<img width="1854" height="1220" alt="Image" src="https://github.com/user-attachments/assets/5ea83067-33b9-409a-8807-e42f727428ab" />

Probably because we are triggering it from a non-svelte file but I don't have time to really investigate.

So for now let's rely on a hack by adding `!important` to CSS classes to override inline styles:

```diff
// src/routes/+layout.svelte
<Toaster
	toastOptions={{
		unstyled: true,
		classes: {
			toast:
-				'shadow-xl px-4 py-3 flex items-center gap-x-3 max-w-full w-full rounded mx-auto text-xs mx-0',
+				'shadow-xl px-4 py-3 flex items-center gap-x-3 max-w-full w-full rounded mx-auto text-xs mx-0 !outline-none',
			loading: 'bg-shade-0',
			error: 'text-red-50 bg-red-700',
			success: 'text-emerald-50 bg-emerald-700',
-			warning: 'text-yellow-50 bg-yellow-700',
+			warning: 'text-yellow-50 !bg-yellow-700',
			info: 'bg-shade-1 text-neutral-500'
		}
	}}
	position="top-center"
/>
```

- Added `!outline-none` to remove blue border
- Added `!bg-yellow-700` to ensure yellow background

mcendon committed:
style: workaround for warning toast styling (#318)
- Added !outline-none to remove blue border
- Added !bg-yellow-700 to ensure yellow background

fmaclen commented on src/routes/+layout.svelte:
```diff
@@ -128,11 +128,11 @@
 			toast:
-				'shadow-xl px-4 py-3 flex items-center gap-x-3 max-w-full w-full rounded mx-auto text-xs mx-0',
+				'shadow-xl px-4 py-3 flex items-center gap-x-3 max-w-full w-full rounded mx-auto text-xs mx-0 !outline-none',
 			loading: 'bg-shade-0',
 			error: 'text-red-50 bg-red-700',
 			success: 'text-emerald-50 bg-emerald-700',
-			warning: 'text-yellow-50 bg-yellow-700',
+			warning: 'text-yellow-50 bg-yellow-700 !bg-yellow-700',
```
> I guess the problem is with the playwright test itself, because when running a local server and manually executes the test in the browser (I am using Chrome 138.0.7204.157 (Official Build) (64-bit) under Pop OS!) the toast styles are fine.

Oh, great catch! I think you are right, I just re-ran the test but took a screenshot instead of using the Trace Viewer and the colors are correct!

We can undo the hacks 👍

fmaclen commented on src/lib/localStorage.ts:
```diff
@@ -16,7 +17,24 @@ function createLocalStorageStore<T>(key: string, defaultValue: T) {
+			} catch (error) {
+				// Handle localStorage quota exceeded error
+				if (error instanceof DOMException && error.name === 'QuotaExceededError') {
+					toast.warning('Localstorage is full', {
```
```suggestion
					toast.warning('Local storage is full', {
```

mcendon replied:
Done

fmaclen commented on tests/session-interaction.test.ts:
```diff
@@ -689,4 +689,45 @@ test.describe('Session interaction', () => {
+		const toastLocator = page.locator('ol[data-sonner-toaster] li[data-type="warning"]');
+		await expect(toastLocator).toBeVisible();
+		await expect(toastLocator).toHaveCount(1);
+		await expect(toastLocator).toContainText('Localstorage is full');
```
```suggestion
		await expect(toastLocator).toContainText('Local storage is full');
```

fmaclen reviewed (changes requested):
Thanks for looking into the Playwright issue. Your earlier implementation appears to be working fine without the hacks.

Also left you 2 minor typo changes.

Other than that, we are good to go!

mcendon committed:
style,test: revert hack on toast styling. improve toast title with @fmaclen suggestion (#318).

mcendon commented:
Perfect Fernando, typo resolved and styling hack was reverted! Great, we are ready to go!

fmaclen commented:
Thanks for the contibution @mcendon!

mcendon commented:
> Thanks for the contribution @mcendon!

You're welcome, Fernando. Count on me for other contributions—I'm short on time, but I'll help however I can! Cheers.
